src/cpp/src/llm_pipeline_static.cpp

```diff
@@ -718,10 +718,17 @@ StatefulLLMPipeline::StatefulLLMPipeline(
         m_kvcache_total = m_max_prompt_len + min_resp_len;
         m_request = compiled.create_infer_request();
     } else {
-        auto model = genai::utils::singleton_core().read_model(models_path / "openvino_model.xml", {}, config);
         ModelConfigDesc model_desc = get_modeldesc_from_json(models_path / "config.json");
         ov::AnyMap properties = config;
-        auto compiled = setupAndCompileModel(model, model_desc, properties);
+        std::shared_ptr<ov::CompiledModel> compiled;
+        // CACHE_DIR + weightless flow support
+        auto cache_mode = get_option<std::string>(config, "CACHE_MODE");
+        if (cache_mode.has_value() && *cache_mode == "OPTIMIZE_SIZE") {
```

Comment:
please, use enum comparison instead of strings

Reply:
Done